### PR TITLE
Fuji subnets

### PR DESCRIPTION
**Pull request by felipemadero**

### CODE REVIEW

File: cmd/control/control.go [hunk 1]
```diff
@@ -52,7 +54,7 @@ func NewCommand() *cobra.Command {
 	cmd.PersistentFlags().StringVar(&logDir, "log-dir", "", "log directory")
 	cmd.PersistentFlags().StringVar(&endpoint, "endpoint", "localhost:8080", "server endpoint")
 	cmd.PersistentFlags().DurationVar(&dialTimeout, "dial-timeout", 10*time.Second, "server dial timeout")
-	cmd.PersistentFlags().DurationVar(&requestTimeout, "request-timeout", 3*time.Minute, "client request timeout")
+	cmd.PersistentFlags().DurationVar(&requestTimeout, "request-timeout", 300*time.Minute, "client request timeout")
```
Comment (felipemadero) [inline cmd/control/control.go:57]: lets keep the default on 3, and let the user specify a special timeout. or... lets decide on 300 minute timeout if fuji

File: local/blockchain.go [hunk 2]
```diff
@@ -93,10 +96,30 @@ func (ln *localNetwork) getNode() node.Node {
 	return node
 }
 
+func (ln *localNetwork) getValidatorWeight() uint64 {
```
Comment (felipemadero) [inline local/blockchain.go:99]: getMinValidatorWeight?

File: local/blockchain.go [hunk 3]
```diff
@@ -93,10 +96,30 @@ func (ln *localNetwork) getNode() node.Node {
...
+		return genesis.LocalParams.MinValidatorStake
+	}
+}
+
 // get node client URI for an arbitrary node in the network
-func (ln *localNetwork) getClientURI() (string, error) { //nolint
+func (ln *localNetwork) getClientURI() (string, error) {
+	clientURI := ""
 	node := ln.getNode()
-	clientURI := fmt.Sprintf("http://%s:%d", node.GetURL(), node.GetAPIPort())
```
Comment (felipemadero) [inline local/blockchain.go:99]: take this back so

File: local/network.go [hunk 4]
```diff
@@ -344,42 +352,54 @@ func NewDefaultConfigNNodes(binaryPath string, numNodes uint32) (network.Config,
...
 				config.HTTPPortKey:    port,
 				config.StakingPortKey: port + 1,
 			},
-			IsBeacon: true,
-		})
+		}
+		if !utils.IsPublicNetwork(networkID) {
+			nodeConfig.IsBeacon = true
+		}
+		/* disable partial sync for now
```
Comment (felipemadero) [inline local/network.go:367]: just remove this ftm

File: server/server.go [hunk 5]
```diff
@@ -49,8 +49,8 @@ const (
 	DefaultNodes uint32 = 5
 
 	stopTimeout           = 30 * time.Second
-	defaultStartTimeout   = 5 * time.Minute
-	waitForHealthyTimeout = 5 * time.Minute
+	defaultStartTimeout   = 300 * time.Minute
```
Comment (felipemadero) [inline server/server.go:52]: this seems to be not used

File: server/server.go [hunk 6]
```diff
@@ -49,8 +49,8 @@ const (
...
+	waitForHealthyTimeout = 300 * time.Minute
```
Comment (felipemadero) [inline server/server.go:53]: can we set this based on network id?

File: utils/constants/constants.go [hunk 7]
```diff
@@ -13,4 +13,5 @@ const (
 	DefaultNetworkID       = 1337
 	DefaultNumNodes        = 5
 	FirstAPIPort           = 9650
+	FujiAPIEndpoint        = "https://api.avax-test.network"
```
Comment (felipemadero) [inline utils/constants/constants.go:16]: not used

**Review by felipemadero — COMMENTED** [review]
just nits

File: local/blockchain.go (hunk 2 above)
Comment (arturrez) [reply local/blockchain.go:99]: fixed

File: local/network.go (hunk 4 above)
Comment (arturrez) [reply local/network.go:367]: done

File: server/server.go (hunk 5 above)
Comment (arturrez) [reply server/server.go:52]: gone now

File: utils/constants/constants.go (hunk 7 above)
Comment (arturrez) [reply utils/constants/constants.go:16]: gone

File: cmd/control/control.go (hunk 1 above)
Comment (arturrez) [reply cmd/control/control.go:57]: k, I can add logic to increase it for fuji

File: server/server.go (hunk 6 above)
Comment (arturrez) [reply server/server.go:53]: added

**Review by arturrez — APPROVED** [review]

**Pull request merged**